data/lua/wml/modify_unit.lua

```diff
@@ -10,20 +10,20 @@ function wml_actions.modify_unit(cfg)
 
 	local replace_mode = cfg.mode == "replace"
 
-	local function handle_attributes(cfg, unit_path, toplevel)
-		for current_key, current_value in pairs(wml.shallow_parsed(cfg)) do
+	local function handle_attributes(cfg2, unit_path, toplevel)
+		for current_key, current_value in pairs(wml.shallow_parsed(cfg2)) do
 			if type(current_value) ~= "table" and (not toplevel or (current_key ~= "type" and current_key ~= "mode")) then
 				wml.variables[string.format("%s.%s", unit_path, current_key)] = current_value
 			end
 		end
 	end
 
-	local function handle_child(cfg, unit_path)
+	local function handle_child(cfg2, unit_path)
 		local children_handled = {}
-		local cfg = wml.shallow_parsed(cfg)
-		handle_attributes(cfg, unit_path)
+		local cfg3 = wml.shallow_parsed(cfg2)
```

Comment:
I don't really like adding numbers just to silence a luacheck warning.

I think `cfg2` and `cfg3` could become the same thing by removing the `local` on this line. As for what to call it… I'm not quite sure. I thought of `sub_cfg` but not entirely sure if that fits.

Reply:
Agreed - I'm also not sure what to call them, but variable names of x1, x2, x3, etc, generally aren't that helpful in all but the most obvious of situations.

Reply:
In this case it's somewhat logical for them to have the same name, as it's a sort of recursive algorithm.

I wonder, do these local functions use anything else from the closure? It's a much bigger change, but maybe they could be extracted out of the main function as local functions at global scope.

Reply:
I took a quick look, and it appears that `handle_attributes` and `handle_child` both don't require any closure and could be moved out of the main function. `handle_unit` however does require closure.

Reply:
Updated.